packages/node-sdk/src/client.ts

```diff
@@ -849,16 +903,24 @@ export class BucketClient {
     };
     checkContextWithTracking(contextWithTracking);
 
-    const params = new URLSearchParams(flattenJSON({ context }));
+    const params = new URLSearchParams(
+      Object.keys(context).length ? flattenJSON({ context }) : undefined,
+    );
+
     if (key) {
       params.append("key", key);
     }
+    const paramsString = params.size ? `?${params.toString()}` : "";
```

Comment:
what does this fix? an empty URLSearchParams returns empty string.

Reply:
Just wanted to avoid having a ? suffix when there's no params. Probably overkill though.